File: backoffice/src/shared/domain/TableId.py
"""
 *
 * Libraries 
 *
"""

from .StringValueObject       import StringValueObject
from .InvalidTableIdException import InvalidTableIdException

"""
 *
 * Classes 
 *
"""

class TableId( StringValueObject ):

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : str ) -> None:
        super().__init__( value )
        if self.isEmpty():
            raise InvalidTableIdException( value )

File: attention/src/shared/table/domain/valueObject/TableId.cs
/* 
 * 
 * Libraries
 *
*/

using attention.src.shared.domain;

/* 
 * 
 * Classes 
 *
*/

namespace attention.src.table.domain {

    public class TableId : UuidValueObject {

        /* 
         * 
         * Methods
         *
        */

        public TableId( string value ) : base( value ) {
            if( isEmpty() ) {
                throw new InvalidTableId( value );
            }
        }

    }

}